itest/oracle_harness.go

```diff
@@ -0,0 +1,313 @@
+package itest
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/rfqmath"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+	oraclerpc "github.com/lightninglabs/taproot-assets/taprpc/priceoraclerpc"
+	"github.com/lightningnetwork/lnd/cert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// oracleHarness is a basic integration test RPC price oracle server harness.
+type oracleHarness struct {
+	oraclerpc.UnimplementedPriceOracleServer
+
+	listenAddr string
+
+	grpcListener net.Listener
+	grpcServer   *grpc.Server
+
+	bidPrices map[string]rfqmath.BigIntFixedPoint
+	askPrices map[string]rfqmath.BigIntFixedPoint
```

Comment:
i think you should add a comment to explain what `string` is here. in other words, I think we should have field level doc comments here (even though it's test code).